core/src/main/java/hudson/Util.java

```diff
@@ -1927,4 +1927,14 @@ public static long daysElapsedSince(@NonNull Date date) {
     private static PathRemover newPathRemover(@NonNull PathRemover.PathChecker pathChecker) {
         return PathRemover.newFilteredRobustRemover(pathChecker, DELETION_RETRIES, GC_AFTER_FAILED_DELETE, WAIT_BETWEEN_DELETION_RETRIES);
     }
+
+    /**
+     * Returns Hex string of SHA-256 Digest of input bytes
+     */
+    public static String getHexStringOfSHA256DigestOf(@NonNull byte[] input) throws NoSuchAlgorithmException
```

Comment:
SHA-256 is mandated by the java specification - so if that is not found there are bigger issues.

it would be better to not make every caller wrap this in a try/catch when if the algorithm is missing something else is wrong.

prefer `NoSuchAlgorithmException` here and throwing a wrapped exception like `IllegalStateException` or `VirtualMachineError`

Reply:
Done. Modified to throw IllegalStateException and logged warning message(which is very unlikely scenario)